adagrams/game.py

```diff
@@ -1,11 +1,151 @@
-    pass
+    #Create a set of 10 letters that matches the letter frequency distribution#
+    draw = []
+    letter_pool_copy = LETTER_POOL.copy()
+    letters_list = list(letter_pool_copy.keys())
+
+    while len(draw) < 10:       
+        #Take a random number within the len of letter pool and get the letter in that index
+        random_letter = letters_list[random.randint(0,len(letter_pool_copy)-1)]
+        
+        #Check if the letter is still availabe in the letter pool and add it to the hand
+        if letter_pool_copy[random_letter] >= 1:
+            draw.append(random_letter)
+            
+            #reduce the frequency in letter pool, for each letter added to the hand
+            letter_pool_copy[random_letter] -= 1         
+    return draw
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #check if the word is an anagram of some or all of the given letters in the hand
+
+    letter_bank_copy = letter_bank.copy()
+    word = word.upper()  
+    for char in word:
+        #Take one of the letters from the hand and make unavailable in the bank copy
+        if char in letter_bank_copy:
+            letter_bank_copy.remove(char)
+        else:
+            return False
 
+    return True
+
+#-------- Score-Word Func Using Nested Lists --------#
+SCORE_CHART = [
+    [1,['A', 'E','I','O','U','L','N','R','S','T']],
+    [2,['D','G']],
+    [3,['B','C','M','P']],
+    [4,['F','H','V','W','Y']],
+    [5,['K']],
+    [8,['J','X']],
+    [10,['Q','Z']]
+]
 def score_word(word):
-    pass
+    #Returns the score of a given word as defined by the SCORE_CHART
+
+    total_points = 0
+    points = 0
+    word = word.upper()
+    for i in range(len(SCORE_CHART)):
+        for char in word:
+            if char in SCORE_CHART[i][1]: #check for char in each letter in list rows
+                points =SCORE_CHART[i][0] #If found, access the index that has the score
+                total_points += points
+            else:
+                points = 0
+
+    #Extra points if word has 7 or more characters 
+    if len(word) >= 7:
+        total_points += 8
+    
+    return total_points
+
+# #------Score_Word Func Using Dictionaries --------#
+# SCORE_CHART = {
+#     1:['A', 'E','I','O','U','L','N','R','S','T'],
+#     2:['D','G'],
+#     3:['B','C','M','P'],
+#     4:['F','H','V','W','Y'],
+#     5:['K'],
+#     8:['J','X'],
+#     10:['Q','Z']
+
+# }
+# def score_word(word):
+#     #Returns the score of a given word as defined by the SCORE_CHART
+
+#     points = 0
+#     word = word.upper()
+
+#     #access the lists in the dict (values)as letters
+#     for score, letters in SCORE_CHART.items():
+#         for i in range(len(letters)): 
+#             for char in word:        
+#                 if char in letters[i]: #check char in
+#                     points += score    
+
+#     #Extra points if word has 7 or more characters 
+#     if len(word) >= 7:
+#         points += 8
+
+#     return points
```

Comment:
I'm sure this is for refactoring reasoning, but don't forget to remove comments like this when pushing up your code. For projects and things it is fine though. More so for industry.